tests/core/usage_statistics/test_usage_statistics_data_context.py

```diff
@@ -0,0 +1,101 @@
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def enable_usage_stats(monkeypatch):
+    monkeypatch.delenv("GE_USAGE_STATS")
+
+
+@pytest.fixture
+def usage_stats_decorated_methods_on_abstract_data_context() -> list[str]:
+    return sorted(
+        map(
+            lambda m: m.split(".")[1],
+            filter(
+                lambda m: m.startswith("AbstractDataContext")
+                and not m.endswith("__init__"),
+                ENABLED_METHODS,
+            ),
+        ),
+    )
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@pytest.mark.parametrize(
+    "data_context_fixture_name",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param("in_memory_runtime_context", id="EphemeralDataContext"),
+        pytest.param("empty_data_context", id="FileDataContext"),
+        pytest.param(
+            "empty_data_context_in_cloud_mode",
+            id="CloudDataContext",
+        ),
+    ],
+)
+@pytest.mark.integration
+def test_all_relevant_context_methods_emit_usage_stats(
+    mock_emit: mock.MagicMock,
+    usage_stats_decorated_methods_on_abstract_data_context: list[str],
+    enable_usage_stats,  # Needs to be before context fixtures to ensure usage stats handlers are attached
+    data_context_fixture_name: str,
+    request,
+):
+    """
+    What does this test and why?
+
+    Ensures that all primary context types (EphemeralDataContext, FileDataContext, and CloudDataContext)
+    emit the same usage stats events.
+
+    This guards against the case where a child class overrides a method defined by AbstractDataContext
+    but forgets to add the decorator, resulting in us losing event data.
+    """
+    context = request.getfixturevalue(data_context_fixture_name)
+
+    relevant_methods = usage_stats_decorated_methods_on_abstract_data_context
+    expected_events = (
+        UsageStatsEvents.DATA_CONTEXT_ADD_DATASOURCE,
+        UsageStatsEvents.DATA_CONTEXT_BUILD_DATA_DOCS,
+        UsageStatsEvents.DATA_CONTEXT_GET_BATCH_LIST,
+        UsageStatsEvents.DATA_CONTEXT_OPEN_DATA_DOCS,
+        UsageStatsEvents.DATA_CONTEXT_RUN_CHECKPOINT,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_ON_DATA,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_WITH_DYNAMIC_ARGUMENTS,
+        UsageStatsEvents.DATA_CONTEXT_RUN_VALIDATION_OPERATOR,
+        UsageStatsEvents.DATA_CONTEXT_SAVE_EXPECTATION_SUITE,
+    )
+    assert len(relevant_methods) == len(
+        expected_events
+    ), "Please update the `expected_events` list to account for all usage stats decorated methods in AbstractDataContext"
+
+    for method_name, expected_event in zip(relevant_methods, expected_events):
```

Comment:
Oh, you do my suggestion here. How do you control the order of `revelevant_methods` and `expected_events` here? It seems like you may want to put them in a dict to more explicitly associate them.